basilmill/util.mill

```diff
@@ -18,6 +18,32 @@ trait DownloadModule extends Module {
   def name: String
   def url: String
   def local: Target[Either[String, String]]
+  def unzip: Option[Unzip] = None
+
+  sealed trait Unzip
+  case class TarGZ(extractedName: String) extends Unzip
+  case class Zip(extractedName: String) extends Unzip
```

Comment:
fileToExtract instead of extractedName? extracted name makes it sound like we're unpacking the zip into the given name.